### PR TITLE
fix types to for better inference + adding response model config updates

### PR DESCRIPTION
**Pull request by roodboi**
Cleaned up types quite a bit - we can now properly get the right inferred return types

Also updated response_model to take an object wiht name and schema - we can extend this and actually use the name - but figured can add those changes in follow up pr as this one is a bit chonky already

### CODE REVIEW

File: docs/blog/posts/anyscale.md [hunk 1]
```diff
@@ -61,7 +61,7 @@ const client = Instructor({
 const user = await client.chat.completions.create({
```
[the remark this answers is not in the document]
Comment (roodboi) [reply docs/blog/posts/anyscale.md:61]: looks that way to me?

[the remark this answers is not in the document]
Comment (roodboi) [reply docs/blog/posts/anyscale.md:61]: oh no - you're right.

[the remark this answers is not in the document]
Comment (jxnl) [reply docs/blog/posts/anyscale.md:61]: @hbrooks it works xD

**Review by jxnl — APPROVED** [review]
looks good, I'll add some documentation on allowing people to set {name} in a seperate pr

**Pull request merged**